src/ast/ast.c

```diff
@@ -527,11 +538,25 @@ char *AST_ToString(const cypher_astnode_t *node) {
 
 	char * str = (char *)cypher_astnode_get_annotation(to_string_ctx, node);
 	if(str == NULL) {
-		struct cypher_input_range range = cypher_astnode_range(node);
-		uint length = range.end.offset - range.start.offset + 2;
-		str = rm_malloc(sizeof(char) * length);
-		strncpy(str, ctx->query_data.query + range.start.offset, length - 1);
-		str[length - 1] = '\0';
+		cypher_astnode_type_t t = cypher_astnode_type(node);
+		const cypher_astnode_t *ast_identifier = NULL;
+		if(t == CYPHER_AST_NODE_PATTERN) {
+			ast_identifier = cypher_ast_node_pattern_get_identifier(node);
+		} else if(t == CYPHER_AST_REL_PATTERN) {
+			ast_identifier = cypher_ast_rel_pattern_get_identifier(node);
+		} else {
+			struct cypher_input_range range = cypher_astnode_range(node);
+			uint length = range.end.offset - range.start.offset + 1;
+			str = rm_malloc(sizeof(char) * length);
+			strncpy(str, ctx->query_data.query_no_params + range.start.offset, length - 1);
+			str[length - 1] = '\0';
+		}
+		if(ast_identifier) {
+			// Graph entity has a user-defined alias, clone it for the annotation.
```

Comment:
This comment is inaccurate, as currently there is no clone and user-defined aliases never get added to annotations.

Reply:
thanks